pmd-core/src/main/java/net/sourceforge/pmd/RuleViolation.java

```diff
@@ -117,4 +119,12 @@
      * @return The variable name.
      */
     String getVariableName();
+
+    /**
+     * Retrieve source code snippet associated with violation.
+     *
+     * @param encoding - source code file encoding
+     * @return {@link SourceCode} - code snippet
+     */
+    SourceCode getSourceCode(String encoding);
```

Comment:
Please don't expose SourceCode here. Your implementation can directly use `getBeginLine` and `getEndLine` and return the sliced string. Also please mark this method as `@Experimental` as we can most likely remove this encoding parameter in pmd 7.

Reply:
replaced SourceCode with String